src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/Integer/ClosedRangeIntegerFrozenSet.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET7_0_OR_GREATER
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+
+namespace System.Collections.Frozen
+{
+    /// <summary>Provides a frozen set to use when the value is an integer, the default comparer is used, and the items are in a contiguous range.</summary>
+    /// <remarks>
+    /// No hashing here, just a simple range check instead.
+    /// </remarks>
+    internal sealed class ClosedRangeIntegerFrozenSet<T> : FrozenSetInternalBase<T, ClosedRangeIntegerFrozenSet<T>.GSW>
+        where T : struct, IBinaryInteger<T>
+    {
+        private readonly T[] _items;
+        private readonly T _min;
+
+        internal ClosedRangeIntegerFrozenSet(T[] entries)
+            : base(EqualityComparer<T>.Default)
+        {
+            Debug.Assert(entries.Length != 0);
+
+            _items = entries;
+            _min = _items[0];
+        }
+
+        private protected override T[] ItemsCore => _items;
+        private protected override Enumerator GetEnumeratorCore() => new Enumerator(_items);
+        private protected override int CountCore => _items.Length;
+
+        private protected override int FindItemIndex(T item)
+        {
+            var index = int.CreateTruncating(item - _min);
+            if ((uint)index < (uint)_items.Length)
+            {
+                return index;
+            }
```

Comment:
Consider the case where `T : long`, `_min = 0`, and `item = (long)(uint.MaxValue) + 1`.  `FindItemIndex` would return the incorrect value `0`. I *think* `CreateSaturating` solves this issue. 

```suggestion
            var index = int.CreateSaturating(item - _min);
            if ((uint)index < (uint)_items.Length)
            {
                return index;
            }
```

Reply:
Unless I'm missing something, I don't think this is an issue. The code in FrozenSet.cs and FrozenDictionary.cs constrains when th the ClosedRange* types are called, it'll never be used with more then 2^31-1 items. Also note that you'd need to have an actual array of longs that big in RAM twice (once as the initial input to the ToFrozenSet call, and once for the array allocated by this collection) in order to get anywhere near this limit.

But maybe I'm missing something...

Reply:
The set would only need to contain one item in this case: `0L`. However `set. Contains((long)(uint.MaxValue) + 1)` would return true!

Reply:
Got it. Fixed and test added.